File: C/PE_7.c
// 10001st prime
//
// https://projecteuler.net/problem=7

#include <math.h>
#include <stdio.h>

int is_prime(int n) {
    int root = sqrt(n);

    for (int i = 2; i <= root; i++) {
        if (n % i == 0) {
            return 0;
        }
    }

    return 1;
}

int main() {
    int n = 1;
    int p_i = 0;

    while (p_i < 10001) {
        n++;
        if (is_prime(n)) {
            p_i++;
        }
    }

    printf("%d\n", n);

    return 0;
}

File: Python/PE-7.py
# 10001st prime
#
# https://projecteuler.net/problem=7

# Querying WolframAlpha with "10001st prime" also works
# https://www.wolframalpha.com/input?i=10001st+prime

from math import sqrt

def isPrime(n):
    for i in range(2, int(sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

primes = []
n = 2

while len(primes) < 10001:
    if isPrime(n):
        primes.append(n)
    n += 1

print(primes[10000])
